api/ephemeral-access/v1alpha1/accessrequest_types.go

```diff
@@ -25,7 +25,7 @@ import (
 
 // Status defines the different stages a given access request can be
 // at a given time.
-// +kubebuilder:validation:Enum=initiated;requested;granted;expired;denied;invalid
+// +kubebuilder:validation:Enum=initiated;requested;granted;expired;denied;invalid;timeout
```

Comment:
Not sure why this makes it back to the status. I expect the resource to be deleted when the TTL has expired. I dont see the point to add this status to the API if it will always gets deleted and the resource wont exist anymore.

Maybe I am missing something?

Reply:
Yes. There are 2 different configurations being introduced in this PR:
1. TTL: Delete the concluded AccessRequests after X amount of time
2. Timeout: Stop the reconciliation and set the AccessRequest status to "timeout" when it fails to get an "approved" response from the plugin after X amount of time.